src/compiler/scala/tools/nsc/typechecker/Namers.scala

```diff
@@ -60,6 +61,11 @@ trait Namers extends MethodSynthesis {
     private lazy val innerNamer =
       if (isTemplateContext(context)) createInnerNamer() else this
 
+    // Cached as a val because `settings.isScala212` parses the Scala version each time...
+    // Not in Namers because then we need to go to outer first to check this.
+    // I do think it's ok to check every time we create a Namer instance (so, not a lazy val).
+    private[this] val isScala212 = settings.isScala212
```

Comment:
We could settings-derived values in `Run` (like we do for `runDefinitions`, which caches member lookups)